src/Configuration/Config.Json/src/JsonStreamConfigurationProvider.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+
+namespace Microsoft.Extensions.Configuration.Json
+{
+    /// <summary>
+    /// Loads configuration key/values from a json stream into a provider.
+    /// </summary>
+    public class JsonStreamConfigurationProvider : StreamConfigurationProvider
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="source">The <see cref="JsonStreamConfigurationSource"/>.</param>
+        public JsonStreamConfigurationProvider(JsonStreamConfigurationSource source) : base(source) { }
+
+        /// <summary>
+        /// Loads json configuration key/values from a stream into a provider.
+        /// </summary>
+        /// <param name="stream">The json <see cref="Stream"/> to load configuration data from.</param>
+        public override void Load(Stream stream)
+        {
+            var data = JsonConfigurationFileParser.Parse(stream);
```

Comment:
Why not set Data directly instead of calling Set?

Reply:
Sure, can switch to setting Data directly, its the same thing

Reply:
Oh yeah you mean just set the dictionary now yeah